### PR TITLE
.NET Core 2.0

### PR DESCRIPTION
**Pull request by BernzSed**
Update to .NET Core 2.0

### CODE REVIEW

File: RunTests.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!
```
Comment (BWortman) [inline RunTests.sh]: This shouldn't be necessary. The chmod is necessary, but not this line change.

Comment (BWortman) [reply RunTests.sh]: I fixed this.

**Comment by BWortman** [comment]
@BernzSed I addressed all comments except for the test errors. Can you please take a look at that?

File: RunTests.sh [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!
```
Comment (BernzSed) [inline RunTests.sh]: "/bin/bash" is necessary for this script to run from a non-bash shell. Using zsh, I can't run this script if "/bin/bash" isn't there.

**Comment by BernzSed** [comment]
Brian,

I'm not able to reproduce the test error. Are you sure you have .NET Core 2.0 installed?

image

**Comment by BWortman** [comment]
Si, I have 2.0 installed:

image

Maybe I'm missing some sort of build step? Is the readme correct? I followed that to set things up, after doing a `git clean -dxf` to ensure I was starting from a clean slate.

**Comment by BernzSed** [comment]
If you're following everything in the readme, I don't think you're missing a build step. I'll keep trying to reproduce the error, but without knowing what's causing it I don't know if I'll get very far. Any information you can find about your environment would be useful.

Could you run `dotnet --info` and send the results?

Also, could you run `dotnet test "./AutoRenter.Api.Tests/AutoRenter.Api.Tests.csproj" -v d` and send the results for that too?

Thanks!

**Comment by BWortman** [comment]
u33400s-MacBook-Pro:autorenter-api-aspnetcore brian$ dotnet --info
.NET Command Line Tools (2.0.0)

Product Information:
 Version:            2.0.0
 Commit SHA-1 hash:  cdcd1928c9

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.12
 OS Platform: Darwin
 RID:         osx.10.12-x64
 Base Path:   /usr/local/share/dotnet/sdk/2.0.0/

Microsoft .NET Core Shared Framework Host

  Version  : 2.0.0
  Build    : e8b8861ac7faf042c87a5c2f9f2d04c98b69f28d

**Comment by BWortman** [comment]
Here's the end (after lots of verbose output) of the dotnet test run:

image

**Comment by BernzSed** [comment]
Try piping the output to a file and upload it.
`dotnet test "./AutoRenter.Api.Tests/AutoRenter.Api.Tests.csproj" -v d > output.txt`

Also, try running it in diagnostic mode and upload the two log files it generates.
`dotnet test "./AutoRenter.Api.Tests/AutoRenter.Api.Tests.csproj" -d diagnostic_output.txt`

**Comment by BWortman** [comment]
Here you go, sir!
[output.txt](https://github.com/fusionalliance/autorenter-api-aspnetcore/files/1420027/output.txt)
[diagnostic_output.host.17-10-26_16-56-58_48289_5.txt](https://github.com/fusionalliance/autorenter-api-aspnetcore/files/1420029/diagnostic_output.host.17-10-26_16-56-58_48289_5.txt)
[diagnostic_output.txt](https://github.com/fusionalliance/autorenter-api-aspnetcore/files/1420030/diagnostic_output.txt)

**Comment by BWortman** [comment]
I added @ToddEJordan, to see if he's having the same problems I'm having...

**Comment by ToddEJordan** [comment]
The tests ran for me on windows under 2.0.2:
C:\autorenter-api-aspnetcore>cd autorenter.api.tests && dotnet test && cd ..\Aut
oRenter.Domain.Services.Tests && dotnet test && cd ..
Build started, please wait...
Build completed.

Test run for C:\autorenter-api-aspnetcore\AutoRenter.Api.Tests\bin\Debug\netcore
app2.0\AutoRenter.Api.Tests.dll(.NETCoreApp,Version=v2.0)
Microsoft (R) Test Execution Command Line Tool Version 15.3.0-preview-20170628-0
2
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
[xUnit.net 00:00:01.1003539]   Discovering: AutoRenter.Api.Tests
[xUnit.net 00:00:01.3387628]   Discovered:  AutoRenter.Api.Tests
[xUnit.net 00:00:01.4703951]   Starting:    AutoRenter.Api.Tests
[xUnit.net 00:00:02.7263824]   Finished:    AutoRenter.Api.Tests

Total tests: 99. Passed: 99. Failed: 0. Skipped: 0.
Test Run Successful.
Test execution time: 5.1216 Seconds
Build started, please wait...
Build completed.

Test run for C:\autorenter-api-aspnetcore\AutoRenter.Domain.Services.Tests\bin\D
ebug\netcoreapp2.0\AutoRenter.Domain.Services.Tests.dll(.NETCoreApp,Version=v2.0
)
Microsoft (R) Test Execution Command Line Tool Version 15.3.0-preview-20170628-0
2
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
[xUnit.net 00:00:00.9004900]   Discovering: AutoRenter.Domain.Services.Tests
[xUnit.net 00:00:01.1283268]   Discovered:  AutoRenter.Domain.Services.Tests
[xUnit.net 00:00:01.2386916]   Starting:    AutoRenter.Domain.Services.Tests
[xUnit.net 00:00:03.2509361]   Finished:    AutoRenter.Domain.Services.Tests

Total tests: 68. Passed: 68. Failed: 0. Skipped: 0.
Test Run Successful.
Test execution time: 5.6162 Seconds

I did not see any log output that looked wrong other than this:
failfail: AutoRenter[0]
      The resizeable api is not registered, this may indicate that you've includ
ed the module but not added the 'ui-grid-resize-columns' directive to your grid
definition.  Cannot raise any events.


I have not looked through Brian's test output yet.

**Comment by ToddEJordan** [comment]
What is ulimit -n?

**Comment by BWortman** [comment]
ulimit: https://www.osc.edu/resources/getting_started/howto/howto_use_ulimit_command_to_set_soft_limits

This is looking like another case of !(works on my box)...

**Comment by BWortman** [comment]
Way to go, @ToddEJordan!!! I increased my ulimit -n from 256 to 512 and everything worked!

**Comment by BWortman** [comment]
Nice work, @BernzSed. And thanks so much for the troubleshooting help, @ToddEJordan.

**Comment by BernzSed** [comment]
Thanks @ToddEJordan! I must have increased my ulimit -n at one time, since it's currently set to 7168.

I'll document this in the readme in case other people have the same problem.

**Comment by BWortman** [comment]
Good idea - tnx, @BernzSed

**Pull request merged**